src/Lean/Message.lean

```diff
@@ -37,16 +38,22 @@ structure NamingContext where
   currNamespace : Name
   openDecls : List OpenDecl
 
+/-- Lazily formatted text to be used in `MessageData`. -/
+structure PPFormat where
+  /-- Pretty-prints text using surrounding context, if any. -/
+  pp : Option PPContext → IO FormatWithInfos
+  /-- Searches for synthetic sorries in original input. Used to filter out certain messages. -/
+  hasSyntheticSorry : MetavarContext → Bool := fun _ => false
```

Comment:
We could probably get rid of this field in favor of "check if `pp` returns a `TermInfo` at the root and use its `expr`" in exchange for formatting terms we will not display